docs/ReleaseProcess.md

```diff
@@ -0,0 +1,64 @@
+# Release Process and Workflow
+
+## Overview
+The Nexus Mods App follows a strict release process which assumes that the
+`main` branch is the development branch and will always be kept in a releasable
+state. This means that any major overhauls of feature development should either
+be done in a single PR or in a be developed behind a optional setting or feature flag. 
+
+At a high level, releases are done by branching from main, and then building a app release
+from this branch. A tag is assigned at the moment of the release. Hotfixes to a release
+will only be done for specific issues that cause major regressions, and these hotfixes will
+be applied to the main branch and backported into the release branch (or vice versa). Hotfix
+releases are branched from the base release. Once it is determined that no hotfixes
+will be applied to the release branch, the release branch is deleted. Often this will be done
+sometime after the next release is made.
+
+## Release Schedule
+The Nexus Mods App developer team follows 3 week iteration cycle, so releases are expected
+to be made at the same interval. Most often releases will be cut by the team lead (halgari)
+who lives in MST timezone. The cutoff point for a release is sometime early morning on Wednesday
+of the release week (MST). Due to the requirement that main kept in a "always releasable" state, 
+the exact time of the creation of the release branch will not be further defined. 
+
+## Hotfix Criteria
+The ultimate goal of the team is to never release a hotfix for a release. However there are times 
+where a regression is discovered at the last moment, or some bug slips in as part of a late PR. These
+situations should be considered the exception not the norm, and common hotfixes should be viewed as 
+a failure of process and addressed accordingly. 
+
+Hotfixes should be restricted to the following criteria:
+
+* Data loss - something in the release will cause the user to lose their data or work in an area of the 
+app that was previously marked as "supported"
+* System instability - something in the app causes crashes in a part of the app outside of the app's influence
+* Inability to upgrade - something in the app causes the app to get into a state where future updates will
+not be installable and the user would be required to wipe their data and start over. 
+
+The common theme for all of these is situations where the user experience is getting worse as time goes on, 
+and waiting till the next release cycle creates an ever compounding problem for users and the dev team. These criteria
+are just guidelines however, and the app team managers reserve the right to make case-by-case decisions. 
+
+## Hotfix development process
+* Bring up an issue with the dev team
+* If a decision is made that the issue needs to be hotfixed, the github issue will be marked as `needs-backporting`
```

Comment:
Can we have a tag for requesting a backport decision? i.e. `requesting-backport-status`. Just so in the event, we don't forget about a potential hotfix over the weekend, etc.